Firestore/Source/Local/FSTMemoryRemoteDocumentCache.mm

```diff
@@ -46,12 +49,12 @@ - (void)addEntry:(FSTMaybeDocument *)document group:(FSTWriteGroup *)group {
   self.docs = [self.docs dictionaryBySettingObject:document forKey:document.key];
 }
 
-- (void)removeEntryForKey:(FSTDocumentKey *)key group:(FSTWriteGroup *)group {
+- (void)removeEntryForKey:(const DocumentKey &)key group:(FSTWriteGroup *)group {
   self.docs = [self.docs dictionaryByRemovingObjectForKey:key];
 }
 
-- (nullable FSTMaybeDocument *)entryForKey:(FSTDocumentKey *)key {
-  return self.docs[key];
+- (nullable FSTMaybeDocument *)entryForKey:(const DocumentKey &)key {
+  return self.docs[(FSTDocumentKey *)key];
```

Comment:
Same comment as before. Avoid c-style casts. (You may be able to rely on the implicit conversion. Otherwise, try static_cast.)